kernel/syscall.c

```diff
@@ -600,6 +602,29 @@ static long _return(long n, long ret)
     return ret;
 }
 
+static int _socketaddr_to_str(
+    const struct sockaddr* addr,
+    char out[],
+    size_t limit)
+{
+    int ret = 0;
+    const uint8_t* p = (uint8_t*)addr->sa_data;
```

Comment:
Maybe add a null check for the addr parameter. Otherwise the related syscalls will crash when passed null address parameters.

Reply:
Good catch. Changed.